bundles/org.openhab.core.model.rule.runtime/src/org/openhab/core/model/rule/runtime/internal/DSLRuleProvider.java

```diff
@@ -88,9 +88,9 @@
 
     private final Logger logger = LoggerFactory.getLogger(DSLRuleProvider.class);
     private final Collection<ProviderChangeListener<Rule>> listeners = new ArrayList<>();
-    private final Map<String, Rule> rules = new HashMap<>();
-    private final Map<String, IEvaluationContext> contexts = new HashMap<>();
-    private final Map<String, XExpression> xExpressions = new HashMap<>();
+    private final Map<String, Rule> rules = new ConcurrentHashMap<>();
+    private final Map<String, IEvaluationContext> contexts = new ConcurrentHashMap<>();
+    private final Map<String, XExpression> xExpressions = new ConcurrentHashMap<>();
     private int triggerId = 0;
     private Set<String> markers = new HashSet<>();
```

Comment:
@kaikreuzer does this need to be thread safe as well?

Reply:
No, afaik this method is called single threaded.